### ISSUE TITLE
Ingressroutes based on old and new CRDs not working in traefik-28.0.0, only regular kubernetes ingress resources are fine

### ISSUE THREAD

**Issue by johanneskastl** [traefik/traefik-helm-chart#1060]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik-helm-chart/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What version of the Traefik's Helm Chart are you using?

traefik-28.0.0

### What version of Traefik are you using?

v3.0.0

### What did you do?

Traefik is installed on all of my k3s cluster via the helm chart, I am not using the traefik included in k3s.

- I read the migration notes, 
- removed all objects based on the old CRDs, 
- removed the old containo.us CRDs,  and 
- applied the definitions for the ingressroutes based on the new CRDs (`traefik.io` instead of `traefik.containo.us`)

I also added some regular Kubernetes ingress resources as a test.

### What did you see instead?

- I see in the Traefik pod logs that Traefik recognizes regular Kubernetes ingress resources (`networking.k8s.io/v1`)
- I see no hint that creating/deleting objects based on the CRDs 
- I can reach things behind regular ingress resources, but not via ingressroutes (no matter if with or without TLS).
- letting the helm chart create a ingressroute for the Traefik Dashboard also does not work, no matter which entrypoint (`traefik`, `web`) I use

### What is your environment & configuration?

Happens in multiple k3s (singlenode) clusters, with e.g. v1.29.4+k3s1 or older versions (1.28.x)

This is the values.yaml used for the installation and the upgrades:

```
ingressClass:
  enabled: true
  isDefaultClass: true
ingressRoute:
  dashboard:
    enabled: false
providers:
  kubernetesCRD:
    allowCrossNamespace: true
  kubernetesIngress:
    publishedService:
      enabled: true
rbac:
  enabled: true
tolerations:
- key: "CriticalAddonsOnly"
  operator: "Exists"
- key: "node-role.kubernetes.io/control-plane"
  operator: "Exists"
  effect: "NoSchedule"
- key: "node-role.kubernetes.io/master"
  operator: "Exists"
  effect: "NoSchedule"
priorityClassName: "system-cluster-critical"
```

### Additional Information

```shell
These are the CRDs present in the cluster:


$ k get crds|grep traefik
ingressroutes.traefik.io                         2024-05-05T08:16:25Z
ingressroutetcps.traefik.io                      2024-05-05T08:16:25Z
ingressrouteudps.traefik.io                      2024-05-05T08:16:25Z
middlewares.traefik.io                           2024-05-05T08:16:26Z
middlewaretcps.traefik.io                        2024-05-05T08:16:26Z
serverstransports.traefik.io                     2024-05-05T08:16:26Z
serverstransporttcps.traefik.io                  2024-05-05T08:16:26Z
tlsoptions.traefik.io                            2024-05-05T08:16:26Z
tlsstores.traefik.io                             2024-05-05T08:16:26Z
traefikservices.traefik.io                       2024-05-05T08:16:26Z
```

I see lots of `reflector.go` errors in the pod logs in one of the clusters, though:

```
W0505 16:36:51.464566       1 reflector.go:539] k8s.io/client-go@v0.29.2/tools/cache/reflector.go:229: failed to list *v1alpha1.ServersTransportTCP: the server could not find the requested resource (get serverstransporttcps.traefik.io)
E0505 16:36:51.464623       1 reflector.go:147] k8s.io/client-go@v0.29.2/tools/cache/reflector.go:229: Failed to watch *v1alpha1.ServersTransportTCP: failed to list *v1alpha1.ServersTransportTCP: the server could not find the requested resource (get serverstransporttcps.traefik.io)

```

**Comment by johanneskastl** [comment]
The reflector.go error seems to be due to #1061

**Comment by mloiseleur** [comment]
An ingressClass is configured in your values:

```yaml
ingressClass:
  enabled: true
  isDefaultClass: true
```

On IngressRoute, documentation on ingressClass [says](https://doc.traefik.io/traefik/providers/kubernetes-crd/#ingressclass):
> If the parameter is set, only resources containing an annotation with the same value are processed. Otherwise, resources missing the annotation, having an empty value, or the value traefik are processed.

Does that answer your question ?

**Comment by mloiseleur** [comment]
Can you provide (complete) reproducible steps for this issue ?

**Comment by johanneskastl** [comment]
> Can you provide (complete) reproducible steps for this issue ?

I wish I could. I do not fully understand what is happening.

The clusters are years old, many many upgrades for Traefik and k3s and the other parts. It might well be that a lot of cruft has been building up.

I got one of the clusters working again by uninstalling and re-installing traefk with the helm chart.

I found I had a deprecated option in a security middleware, but that was not used in the ingressroutes, so that should not have mattered.

It might have been the missing CRD (see #1061) that tripped the Traefik proxy.

I'll try to dig deeper and report back my findings. If I am the only one bitten by this, then it might well be some constellation in my clusters.
